scripts/test/spec_helper.rb

```diff
@@ -0,0 +1,180 @@
+# File managed by pluginsync
+
+require 'hashie'
+require 'json'
+require 'pathname'
+require 'yaml'
+require 'rspec/retry'
+require 'dockerspec/serverspec'
+
+begin
+  require 'pry'
+rescue LoadError
+end
+
+module SnapUtils
+  def sh(arg)
+    c = command(arg)
+    puts c.stderr
+    puts c.stdout
+  end
+
+  def build_path
+    File.expand_path(File.join(__FILE__, '../../../build/linux/x86_64'))
+  end
+
+  def local_plugins
+    Dir.chdir(build_path)
+    @local_plugins ||= Dir.glob("snap-plugin-*")
+  end
+
+  def load_plugin(type, name, version="latest")
+    plugin_name = "snap-plugin-#{type}-#{name}"
+
+    # NOTE: use mock2 plugin when mock is requested, in general we should avoid mock collector.
+    case name
+    when 'mock' # TODO: revisit how we handle mock plugins
+      url = "https://s3-us-west-2.amazonaws.com/snap.ci.snap-telemetry.io/snap/#{version}/linux/x86_64/#{plugin_name}2"
+    when 'mock1', 'mock2', 'mock2-grpc', 'passthru', 'passthru-grpc', 'mock-file', 'mock-file-grpc'
+      url = "https://s3-us-west-2.amazonaws.com/snap.ci.snap-telemetry.io/snap/#{version}/linux/x86_64/#{plugin_name}"
+    else
+      url = "https://s3-us-west-2.amazonaws.com/snap.ci.snap-telemetry.io/plugins/#{plugin_name}/#{version}/linux/x86_64/#{plugin_name}"
+    end
+
+    if local_plugins.include? plugin_name
+      command("snaptel plugin load #{build_path}/#{plugin_name}").exit_status
+    else
+      command("curl -sfL #{url} -o /opt/snap/plugins/#{plugin_name}").exit_status
+      command("snaptel plugin load /opt/snap/plugins/#{plugin_name}").exit_status
+    end
+  end
+
+  def cmd_with_retry(arg, opt={ :timeout => 30 })
+    cmd = command(arg)
+    while cmd.exit_status != 0 or cmd.stdout == '' and opt[:timeout] > 0
+      sleep 5
+      opt[:timeout] -= 5
+      cmd = command(arg)
+    end
+    return cmd
+  end
+
+  def curl_json_api(url)
+    output = cmd_with_retry("curl #{url}").stdout
+    if output.size > 0
+      JSON.parse(output)
+    else
+      {}
+    end
+  end
+
+  def load_json(file)
```

Comment:
This is part of pluginsync framework, if we want to modify this we should send a PR and change it in the pluginsync repo.